src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/LoopDetector.kt

```diff
@@ -222,6 +249,68 @@ internal class LoopDetector(
             Decision.Idle
     }
 
+    /**
+     * Called before regular method calls.
+     */
+    fun beforeRegularMethodCall(codeLocation: Int, params: Array<Any?>) {
```

Comment:
Maybe just `beforeMethodCall`, because:
1. there is no other kinds of method calls in the `LoopDetector` API
2. to keep it in-sync with the naming scheme in `ManagedStrategy` (which uses names `beforeMethodCall` and `beforeAtomicMethodCall`)